src/wp-includes/blocks.php

```diff
@@ -1332,3 +1332,29 @@ function _wp_multiple_block_styles( $metadata ) {
 	return $metadata;
 }
 add_filter( 'block_type_metadata', '_wp_multiple_block_styles' );
+
+/**
+ * This function takes care of adding inline styles
+ * in the proper place, depending on the theme in use.
+ *
+ * For block themes, it's loaded in the head.
+ * For classic ones, it's loaded in the body
+ * because the wp_head action (and wp_enqueue_scripts)
+ * happens before the render_block.
+ *
+ * See https://core.trac.wordpress.org/ticket/53494.
+ *
+ * @param string $style String containing the CSS styles to be added.
+ */
+function wp_enqueue_block_support( $style ) {
```

Comment:
1) The function name does not fully identify what happens when this function is invoked. It's purpose is to enqueue styles rather than "support". How about `wp_enqueue_inline_block_styles` ? @oandregal what do you think?

2) Let's move this function to `wp-includes/script-loader.php` where like functions exist.

3) Nitpick: `@since` annonation is needed (though can be added by core committer in prepping the commit.

Reply:
1. Naming. I'd like to retain the `block_support` if possible, perhaps `wp_enqueue_block_support_styles`?
2. Moving: I'm fine with that place if folks are opinionated about that. I also wanted to share more of the rationale as to why `blocks.php` sounds like a good place as well: there're functions such as `wp_enqueue_block_style` or `wp_migrate_old_typography_shape` that deal with block things (styles, block supports, etc).
3. Sorry I've missed the `@since` tag.